--- a/drivers/net/wireless/realtek/rtkwifiu/rtl8814ae/os_dep/linux/ioctl_cfg80211.c
+++ b/drivers/net/wireless/realtek/rtkwifiu/rtl8814ae/os_dep/linux/ioctl_cfg80211.c
@@ -6084,6 +6084,33 @@ void rtw_cfg80211_init_wdev_data(_adapter *padapter)
 #endif
 }
 
+#if defined(CONFIG_80211AC_VHT) && (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 6, 0))
+static void rtw_cfg80211_init_vht_capab(_adapter *padapter
+	, struct ieee80211_sta_vht_cap *sta_vht_cap, BAND_TYPE band, u8 rf_type)
+{
+	struct hal_spec_t *hal_spec = GET_HAL_SPEC(padapter);
+	struct registry_priv *regsty = &padapter->registrypriv;
+	struct vht_priv *vhtpriv = &padapter->mlmepriv.vhtpriv;
+	u8 vht_cap_ie[2 + 12] = {0};
+	u8 bw;
+
+	if (!REGSTY_IS_11AC_ENABLE(regsty)
+		|| !hal_chk_proto_cap(padapter, PROTO_CAP_11AC)
+	) {
+		sta_vht_cap->vht_supported = 0;
+		return;
+	}
+
+	rtw_vht_use_default_setting(padapter);
+	rtw_build_vht_cap_ie(padapter, vht_cap_ie);
+
+	sta_vht_cap->vht_supported = 1;
+
+	_rtw_memcpy(&sta_vht_cap->cap, vht_cap_ie + 2, 4);
+	_rtw_memcpy(&sta_vht_cap->vht_mcs, vht_cap_ie + 2 + 4, 8);
+}
+#endif /* defined(CONFIG_80211AC_VHT) && (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 6, 0)) */
+
 void rtw_cfg80211_init_wiphy(_adapter *padapter)
 {
 	u8 rf_type;
@@ -6103,8 +6130,12 @@ void rtw_cfg80211_init_wiphy(_adapter *padapter)
 #ifdef CONFIG_IEEE80211_BAND_5GHZ
 	if (IsSupported5G(padapter->registrypriv.wireless_mode)) {
 		bands = wiphy->bands[IEEE80211_BAND_5GHZ];
-		if (bands)
+		if (bands) {
 			rtw_cfg80211_init_ht_capab(padapter, &bands->ht_cap, IEEE80211_BAND_5GHZ, rf_type);
+			#if defined(CONFIG_80211AC_VHT) && (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 6, 0))
+			rtw_cfg80211_init_vht_capab(padapter, &bands->vht_cap, BAND_ON_5G, rf_type);
+			#endif
+		}
 	}
 #endif
 	/* init regulary domain */
