--- a/drivers/net/wireless/realtek/rtkwifiu/rtl8723bs/os_dep/linux/os_intfs.c
+++ b/drivers/net/wireless/realtek/rtkwifiu/rtl8723bs/os_dep/linux/os_intfs.c
@@ -2544,7 +2544,7 @@ int _netdev_vir_if_open(struct net_device *pnetdev)
 #endif
 
 #ifdef CONFIG_IOCTL_CFG80211
-	rtw_cfg80211_init_wiphy(padapter);
+	/* rtw_cfg80211_init_wiphy(padapter); */
 	rtw_cfg80211_init_wdev_data(padapter);
 #endif
 
@@ -2996,6 +2996,9 @@ int rtw_os_ndevs_register(struct dvobj_priv *dvobj)
 				rtw_warn_on(1);
 				break;
 			}
+#ifdef CONFIG_IOCTL_CFG80211
+			rtw_cfg80211_init_wiphy(adapter);
+#endif
 		}
 	}
 
@@ -3188,7 +3191,7 @@ int _netdev_open(struct net_device *pnetdev)
 #endif /* !RTW_HALMAC */
 
 #ifdef CONFIG_IOCTL_CFG80211
-		rtw_cfg80211_init_wiphy(padapter);
+		/* rtw_cfg80211_init_wiphy(padapter); */
 		rtw_cfg80211_init_wdev_data(padapter);
 #endif
 
