--- a/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/os_dep/linux/os_intfs.c
+++ b/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/os_dep/linux/os_intfs.c
@@ -2254,7 +2254,7 @@ int _netdev_vir_if_open(struct net_device *pnetdev)
 #endif
 
 #ifdef CONFIG_IOCTL_CFG80211
-		rtw_cfg80211_init_wiphy(padapter);
+		/* rtw_cfg80211_init_wiphy(padapter); */
 		rtw_cfg80211_init_wdev_data(padapter);
 #endif
 
@@ -2577,6 +2577,9 @@ int rtw_os_ndevs_register(struct dvobj_priv *dvobj)
 				rtw_warn_on(1);
 				break;
 			}
+#ifdef CONFIG_IOCTL_CFG80211
+			rtw_cfg80211_init_wiphy(adapter);
+#endif
 		}
 	}
 
@@ -2765,7 +2768,7 @@ int _netdev_open(struct net_device *pnetdev)
 		rtw_intf_start(padapter);
 
 #ifdef CONFIG_IOCTL_CFG80211
-		rtw_cfg80211_init_wiphy(padapter);
+		/* rtw_cfg80211_init_wiphy(padapter); */
 		rtw_cfg80211_init_wdev_data(padapter);
 #endif
 
