--- a/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Kconfig
+++ b/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Kconfig
@@ -1,6 +1,6 @@
-config RTL8822BU
-	tristate "Realtek 8822B USB WiFi"
+config RTL8812BU
+	tristate "Realtek 8812B USB WiFi"
 	depends on USB
 	---help---
-	  Help message of RTL8822BU
+	  Help message of RTL8812BU
 
--- a/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Makefile
+++ b/drivers/net/wireless/realtek/rtkwifiu/rtl8812bu/Makefile
@@ -1930,11 +1930,11 @@ ifeq ($(CONFIG_RTL8723B), y)
 $(MODULE_NAME)-$(CONFIG_MP_INCLUDED)+= core/rtw_bt_mp.o
 endif
 
-obj-$(CONFIG_RTL8822BU) := $(MODULE_NAME).o
+obj-$(CONFIG_RTL8812BU) := $(MODULE_NAME).o
 
 else
 
-export CONFIG_RTL8822BU = m
+export CONFIG_RTL8812BU = m
 
 all: modules
 
