--- a/drivers/net/wireless/realtek/rtkwifiu/rtl8821cs/os_dep/linux/os_intfs.c
+++ b/drivers/net/wireless/realtek/rtkwifiu/rtl8821cs/os_dep/linux/os_intfs.c
@@ -2668,7 +2668,7 @@ int _netdev_vir_if_open(struct net_device *pnetdev)
 #endif
 
 #ifdef CONFIG_IOCTL_CFG80211
-	rtw_cfg80211_init_wiphy(padapter);
+	/* rtw_cfg80211_init_wiphy(padapter); */
 	rtw_cfg80211_init_wdev_data(padapter);
 #endif
 
@@ -3124,6 +3124,9 @@ int rtw_os_ndevs_register(struct dvobj_priv *dvobj)
 				rtw_warn_on(1);
 				break;
 			}
+#ifdef CONFIG_IOCTL_CFG80211
+			rtw_cfg80211_init_wiphy(adapter);
+#endif
 		}
 	}
 
@@ -3314,7 +3317,7 @@ int _netdev_open(struct net_device *pnetdev)
 		#endif
 
 		#ifdef CONFIG_IOCTL_CFG80211
-		rtw_cfg80211_init_wiphy(padapter);
+		/* rtw_cfg80211_init_wiphy(padapter); */
 		rtw_cfg80211_init_wdev_data(padapter);
 		#endif
 		/* rtw_netif_carrier_on(pnetdev); */ /* call this func when rtw_joinbss_event_callback return success */
@@ -3425,7 +3428,7 @@ int _netdev_open(struct net_device *pnetdev)
 #endif /* !RTW_HALMAC */
 
 #ifdef CONFIG_IOCTL_CFG80211
-		rtw_cfg80211_init_wiphy(padapter);
+		/* rtw_cfg80211_init_wiphy(padapter); */
 		rtw_cfg80211_init_wdev_data(padapter);
 #endif
 
